packages/neon-api/__tests__/transaction/validator.ts

```diff
@@ -0,0 +1,320 @@
+import {
+  TransactionBuilder,
+  TransactionValidator,
+  ValidationAttributes
+} from "../../src/transaction";
+import { sc, tx, u, rpc, CONST } from "@cityofzion/neon-core";
+
+const TESTNET_URL = "http://seed5t.neo.org:20332";
+const rpcClient = new rpc.RPCClient(TESTNET_URL);
+
+const sig = {
+  invocationScript:
+    "40f52d1206315dfac64c14ec2dfef1edd62f4460487c23be6bbbbf9080973784ca7dbfe4dfcf4b6b82f2921b968e0d693020b76be0b20171ac56e7da50ab1c4b06",
+  verificationScript:
+    "21031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c968747476aa"
+};
+const multiSig = {
+  invocationScript:
+    "40f52d1206315dfac64c14ec2dfef1edd62f4460487c23be6bbbbf9080973784ca7dbfe4dfcf4b6b82f2921b968e0d693020b76be0b20171ac56e7da50ab1c4b0640efe3ccf3a49dd670d8785a12218324f60a6b56ed5a628f15522b883a81b51ea9256c0be62008377b156eb1a76e6dc25aad776524c18eb01b0810ed833b15a1ca",
+  verificationScript:
+    "5221031d8e1630ce640966967bc6d95223d21f44304133003140c3b52004dc981349c92103767002bb9f74317035ce8d557a3aed30ce831eb16b5f636a139dad0b07916bed210329898e6e5e0a2f175e205b4019c500d6bb69203b56470ec2fc8ab0a4c065e16d5368c7c34cba"
+};
+
+describe("validateValidUntilBlock", () => {
+  test("valid", async () => {
+    const transaction = new TransactionBuilder({
+      validUntilBlock:
+        tx.Transaction.MAX_TRANSACTION_LIFESPAN +
+        (await rpcClient.getBlockCount()) -
+        1
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateValidUntilBlock();
+    expect(result.valid).toBeTruthy();
+  });
+
+  test("invalid", async () => {
+    const transaction = new TransactionBuilder({
+      validUntilBlock: 1
+    }).build();
+    const validator = new TransactionValidator(rpcClient, transaction);
+    const result = await validator.validateValidUntilBlock();
+    expect(result.valid).toBeFalsy();
```

Comment:
Not checking the other fields?

Reply:
Check the whole object when autoFix = true. Here we only care about whether it's valid